src/operator/every.ts

```diff
@@ -8,38 +8,41 @@ import {Subscriber} from '../Subscriber';
 import {tryCatch} from '../util/tryCatch';
 import {errorObject} from '../util/errorObject';
 
+export type Predicate<T> = (value: T, index: number, source: Observable<T>) => boolean;
+
 /**
  * Returns an Observable that emits whether or not every item of the source satisfies the condition specified.
  * @param {function} predicate a function for determining if an item meets a specified condition.
  * @param {any} [thisArg] optional object to use for `this` in the callback
  * @returns {Observable} an Observable of booleans that determines if all items of the source Observable meet the condition specified.
  */
-export function every<T>(predicate: (value: T, index: number, source: Observable<T>) => boolean,
-                         thisArg?: any): Observable<boolean> {
-  const source = this;
-  if (source._isScalar) {
-    const result: boolean = tryCatch(predicate).call(thisArg || this, source.value, 0, source);
+export function every<T>(predicate: Predicate<T>, thisArg?: any): Observable<boolean> {
+    
+  let _this: Observable<T> = this;
+  if (_this._isScalar) {
+    let scalar: ScalarObservable<T> = <any>_this;
```

Comment:
It's true that there's no direct benefit to end users for this kind of change, but since I was doing typing fixes in this PR, I also tried to prevent mistakes in typings in the future. The `isEmpty` operator you commented on earlier is a good example of this. You're trying to take an `Observable<any>` and produce an `Observable<boolean>`, but you're using an `Operator<boolean, boolean>` to do it, which the type system would disqualify. If you cast `this` to `(<Observable<T>>this)`, the compiler complains about how the operator has the wrong type, which has the knock on effect of forcing you to fix your types all the way down to `IsEmptySubscriber`.

In and of itself this has no effect on the end user, but in many cases the user facing type signature is wrong simply because of a break in type checking somewhere down the line. It's also true that things "just work" because the people that have been contributing so far understand the codebase well, and you have good unit tests to cover your work. In some cases people might not have as good an understanding of the overall codebase (yours truly is a good example), and its nice to have accurate static typing to hold your hand.